README.md

```diff
@@ -203,14 +200,15 @@ void StartActivity()
 
     // extract other things from Activity and set them on span (tags to attributes)
     // ...
-
-    tracer.WithSpan(span); // we drop scope here as we cannot propagate it all the way to stop event
 }
 
 void StopActivity()
 {
     var span = tracer.CurrentSpan;
-    span.End();
+    if (span is IDisposable disposableSpan)

```

Comment:
this is bad to advertise a type casting.

Reply:
so IDisposable ISpan? Any objections?

I'm strongly against another Scope type that has span and is IDisposable

Reply:
alternative is to make `span.End()` restore the parent's context if span was started with `StartActiveSpan`. Which may or may not be a better approach... It kind of what will happen when Span will become IDisposable. `End`  method semantics will change depending on how it was started and whether it was associated with the scope